stestr/tests/test_return_codes.py

```diff
@@ -199,3 +204,48 @@ def test_load_from_stdin(self):
         stream = self._get_cmd_stdout(
             'stestr last --subunit')[0]
         self.assertRunExit('stestr load', 0, stdin=stream)
+
+    def test_parallel_passing_from_func(self):
+        self.assertEqual(0, run.run_command(filters=['passing']))
+
+    def test_parallel_passing_bad_regex_from_func(self):
+        self.assertEqual(1, run.run_command(filters=['bad.regex.foobar']))
+
+    def test_parallel_fails_from_func(self):
+        self.assertEqual(1, run.run_command())
+
+    def test_serial_passing_from_func(self):
+        self.assertEqual(0, run.run_command(filters=['passing'], serial=True))
+
+    def test_serial_fails_from_func(self):
+        self.assertEqual(1, run.run_command(serial=True))
```

Comment:
https://travis-ci.org/mtreinish/stestr/jobs/288739848#L673
I'm actually not sure, but this should be like this?
```
stdout = io.BytesIO()
self.assertEqual(1, run.run_command(serial=True, stdout=stdout))
```

Reply:
ah, this causes another issue... :(
```
TypeError: 'unicode' does not have the buffer interface
```

Reply:
yeah, I've got a branch locally that adds the stdout redirect to everything. I have to debug it some more to work through the unicode errors. But I decided to wait on that until the new subunit release is out, because we're blocked on that anyway.